### PR TITLE
[WIP] remove mlab from tutorials

### PR DESCRIPTION
**Pull request by agramfort**
cc @GuillaumeFavelier 

remaining ones are related to plotting of dipoles

### CODE REVIEW

File: tutorials/source-modeling/plot_source_alignment.py [hunk 1]
```diff
@@ -64,11 +63,11 @@
...
-mne.viz.plot_alignment(raw.info, trans=trans, subject='sample',
-                       subjects_dir=subjects_dir, surfaces='head-dense',
-                       show_axes=True, dig=True, eeg=[], meg='sensors',
-                       coord_frame='meg')
-mlab.view(45, 90, distance=0.6, focalpoint=(0., 0., 0.))
+fig = mne.viz.plot_alignment(raw.info, trans=trans, subject='sample',
+                             subjects_dir=subjects_dir, surfaces='head-dense',
+                             show_axes=True, dig=True, eeg=[], meg='sensors',
+                             coord_frame='meg')
+mne.viz_3d_view(fig, 45, 90, distance=0.6, focalpoint=(0., 0., 0.))
```
Comment (GuillaumeFavelier) [inline tutorials/source-modeling/plot_source_alignment.py]: Here it's `mne.viz.set_3d_view()`. That's why [circle](https://circleci.com/gh/mne-tools/mne-python/13866?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) is complaining.

**Comment by GuillaumeFavelier** [comment]
very good start @agramfort , I'll take care of the plotting of dipoles then.

**Comment by agramfort** [comment]
I would add an option to "mode" parameter to Dipole.plot_locations to
support plotting dipoles on the output of a plot_alignment figure

You'll need maybe to add a figure parameter and a color parameter

**Comment by GuillaumeFavelier** [comment]
I added the skeleton for `plot_dipole_locations()` and updated the parameters accordingly but I don't know how to get/set the dipoles position of the `Dipole`/`DipoleFixed` class correctly. Do we need to `apply_trans()` for example?

**Comment by larsoner** [comment]
Positions of `Dipole` are (or should be) always in the head coordinate frame. This should be enough for you to figure out whether or not you need to transform them based on which coordinate frame you are plotting in, MEG/head/mri.

**Comment by larsoner** [comment]
Please add a note to the `Dipole` / `DipoleFixed` class to say this so that the next dev/user is not confused

**Comment by agramfort** [comment]
@GuillaumeFavelier you need to factor out the code to get dipole_lo, cs from:

https://github.com/mne-tools/mne-python/blob/6582da009a174ebca97b7f67bef65e9c3ea5ddf6//mne/viz/_3d.py#L2708

```
def _get_dipole_loc(dip, coord_frame, trans):
     ....
     return dipole_locs, dipole_oris
```

then make sure `def _plot_dipole_mri_orthoview` uses this private function

File: mne/viz/_3d.py [hunk 2]
```diff
@@ -2411,8 +2411,8 @@ def plot_sparse_source_estimates(src, stcs, colors=None, linewidth=2,
 @verbose
 def plot_dipole_locations(dipoles, trans, subject, subjects_dir=None,
                           mode='orthoview', coord_frame='mri', idx='gof',
-                          show_all=True, ax=None, block=False,
-                          show=True, verbose=None):
+                          show_all=True, ax=None, block=False, show=True,
+                          verbose=None, color=(1.0, 0.0, 0.0), fig=None):
```
Comment (agramfort) [inline mne/viz/_3d.py]: keep verbose at the end

File: mne/viz/_3d.py [hunk 3]
```diff
@@ -2489,6 +2494,17 @@ def plot_dipole_locations(dipoles, trans, subject, subjects_dir=None,
...
+    elif mode in ['sphere', 'cone']:
+        from .backends.renderer import _Renderer
+        renderer = _Renderer(fig=fig, size=(600, 600))
+
+        pos, ori, _, _ = _get_dipole_loc(dipoles, trans, subject,
+                                         subjects_dir=subjects_dir,
+                                         coord_frame=coord_frame)
+
+        renderer.sphere(center=np.c_[pos[:, 0], pos[:, 1], pos[:, 2]],
+                        color=color, scale=1E-3)
```
Comment (agramfort) [inline mne/viz/_3d.py]: dipoles are not expect to be sphere. They have a position and an orientation. It's better to use Arrows like:

https://mne-tools.github.io/dev/auto_tutorials/sample-datasets/plot_brainstorm_phantom_elekta.html?highlight=plot_brainstorm_phantom_elekta

Comment (GuillaumeFavelier) [reply mne/viz/_3d.py]: Some tutorials were more interested in the locations only actually.

**Comment by GuillaumeFavelier** [comment]
Waiting for Circle now to share some renderings.

**Comment by GuillaumeFavelier** [comment]
The last use of `mlab` is in [plot_dipole_orientations.py](https://github.com/mne-tools/mne-python/blob/2acefbf35a69dc873d64e155b1e888cf257c4d13/tutorials/source-modeling/plot_dipole_orientations.py#L160-L162). Can we take care of this one in a follow-up PR? We may need to discuss how to handle this particular case.

image

File: mne/viz/_3d.py (hunk 3 above)
Comment (GuillaumeFavelier) [reply mne/viz/_3d.py]: Like this one: `plot_phantom_4DBTi.py`

Comment (GuillaumeFavelier) [reply mne/viz/_3d.py]: And the first plot of `plot_dipole_orientations.py` I think.

File: tutorials/sample-datasets/plot_brainstorm_phantom_elekta.py [hunk 4]
```diff
@@ -151,29 +151,25 @@
...
-    mlab.points3d(x, y, z, scale_factor=0.005, opacity=opacity, color=color)
-    q = mlab.quiver3d(x, y, z, u, v, w,
-                      scale_factor=0.03, opacity=opacity,
-                      color=color, mode='arrow')
-    q.glyph.glyph_source.glyph_source.shaft_radius = 0.02
-    q.glyph.glyph_source.glyph_source.tip_length = 0.1
-    q.glyph.glyph_source.glyph_source.tip_radius = 0.05
-
+actual_amp = np.ones(len(dip))  # misc amp to create Dipole instance
+actual_gof = np.ones(len(dip))  # misc amp to create Dipole instance
```
Comment (agramfort) [inline tutorials/sample-datasets/plot_brainstorm_phantom_elekta.py]:
```suggestion
actual_gof = np.ones(len(dip))  # misc GOF to create Dipole instance
```

File: tutorials/sample-datasets/plot_phantom_4DBTi.py [hunk 5]
```diff
@@ -64,15 +65,16 @@
 
 ###############################################################################
 # Plot the dipoles in 3D
+actual_amp = np.ones(len(dip))  # misc amp to create Dipole instance
+actual_gof = np.ones(len(dip))  # misc amp to create Dipole instance
```
Comment (agramfort) [inline tutorials/sample-datasets/plot_phantom_4DBTi.py]:
```suggestion
actual_gof = np.ones(len(dip))  # misc GOF to create Dipole instance
```

**Comment by agramfort** [comment]
besides LGTM

+1 for take care of dip oris in src in a next short PR

**Comment by larsoner** [comment]
> +1 for take care of dip oris in src in a next short PR

Sure but for now the changes to any broken example should be reverted.

@GuillaumeFavelier can you look at all changed examples on `master` and make sure they are still okay on this PR? They don't have to be identical, but they can't for example change from what [we currently have in this example](http://mne-tools.github.io/dev/auto_tutorials/source-modeling/plot_dipole_orientations.html#the-source-space) in `master`:



To [what's in this PR](https://13916-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/source-modeling/plot_dipole_orientations.html#the-source-space):

**Comment by GuillaumeFavelier** [comment]
I'm on it!

**Comment by GuillaumeFavelier** [comment]
I think it should be good enough now, what do you think @larsoner ?

File: tutorials/sample-datasets/plot_phantom_4DBTi.py [hunk 6]
```diff
@@ -73,8 +73,8 @@
 fig = mne.viz.plot_alignment(evoked.info, bem=sphere, surfaces=[])
 
 # Plot the position of the actual dipole
-fig = mne.viz.plot_dipole_locations(dipoles=dip_true, mode='3d',
-                                    color=(1., 0., 0.), fig=fig)
+fig = mne.viz.plot_dipole_locations(dipoles=dip_true, mode='sphere',
+                                    scale=6e-3, color=(1., 0., 0.), fig=fig)
```
Comment (larsoner) [inline tutorials/sample-datasets/plot_phantom_4DBTi.py]: new `scale` should not be required

File: mne/viz/_3d.py [hunk 7]
```diff
@@ -2506,11 +2509,12 @@ def plot_dipole_locations(dipoles, trans=None, subject=None, subjects_dir=None,
             pos = apply_trans(trans, pos)
             ori = apply_trans(trans, ori)
 
-        renderer.sphere(center=pos, color=color, scale=5e-3)
-        x, y, z = pos.T
-        u, v, w = ori.T
-        renderer.quiver3d(x, y, z, u, v, w, scale=0.03,
-                          color=color, mode='arrow')
+        renderer.sphere(center=pos, color=color, scale=scale)
```
Comment (larsoner) [inline mne/viz/_3d.py]: for backward compat `scale` should default to `5e-3` instead of `1.`

**Comment by GuillaumeFavelier** [comment]
I obtain this on my machine for the last figure:

image

I will force a `mlab.draw()` in `set_3d_view()` just in case.

**Comment by GuillaumeFavelier** [comment]
I ran the tutorial locally with `PATTERN=plot_brainstorm_phantom_elekta make html_dev-pattern` and I didn't reproduce this behaviour. This is what I obtain: 

image

What about the result of the last `[circle full]` I did?

**Comment by larsoner** [comment]
Thanks @agramfort @GuillaumeFavelier

**Pull request merged**